### ISSUE TITLE
train issue

### ISSUE THREAD

**Issue by m0nesy-111** [arpanpoudel/lenslessimaging#4]
Thank you for your code，I conducted an experiment on lensless imaging and encountered an issue. Theoretically, the reconstructed result should display the target image on a black background, but due to camera alignment, the final image may not be centered in the reconstruction result, and it might even appear tilted. I would like to ask if you have any good methods for creating the ‘’ground truth‘’.Thank you again

**Comment by arpanpoudel** [comment]
Thank you for your question! We encountered a similar issue with misalignment in the lensless imaging dataset (Waller-Lab-LenslessLearning), where both the ground truth (GT) images and lensless reconstructions had a consistent shift. Since we used the Point Spread Function (PSF/H) from the original dataset, we trained our model using the shifted images.
To handle this misalignment in post-processing, we applied the following corrections to both the reconstructed images and ground truth:
```python
 def preplot(image):
    # Convert from CHW to HWC for matplotlib
    image = torch.tensor(image)

    # Change CHW to HWC
    image = torch.permute(image, (1, 2, 0))

    # Convert BGR to RGB
    image = image[:, :, [2, 1, 0]]

    # Clip values to 0-1 range
    out = torch.clip(image, 0, 1)

    # Flip the image upside down
    out = torch.flipud(out)

    # Crop the image to correct the misalignment
    return out[60:, 62:-38, :]

This is the straightforward approach when all the images have a similar shift.

**Comment by m0nesy-111** [comment]
Thank you for your patient response. Due to the "alignment" issue, there may be anomalies during the training of Unet. Also, were the values [60:, 62:-38, :] obtained through experimentation? Is the cropping of pixel values done to ensure training stability? Is it necessary to standardize and normalize both the reconstructed images and the ground truth? (My ADMM and Unet are trained separately, without using the EnsembleModel.)

**Comment by arpanpoudel** [comment]
I do not think there will be "anomalies" during training because every ground-truth and corresponding lensless images have the same shift. Therefore, Unet will learn such a shift. You can verify by modifying the preplot function to just plot all pixels without cropping (eg, return output instead of `return out[60:, 62:-38, :]`). Those crop values were obtained from the original dataset, such that we do not show the out-of-frame pixel value. Remember that cropping is done at post-processing (meaning the model reconstructs the shifted image and we crop the out of frame pixels). This is necessary because we are using a PSF calibrated for such system.

**Comment by m0nesy-111** [comment]
I understand now. I misunderstood your point earlier. Thank you again for your explanation.
